authenticate-with-oauth/src/main/java/com/esri/arcgismaps/sample/authenticatewithoauth/MainActivity.kt

```diff
@@ -17,103 +17,50 @@
-        )[OAuthUserSignInViewModel::class.java]
-
-        lifecycle.addObserver(oAuthUserSignInViewModel)
-        lifecycle.addObserver(mapView)
-
-        setUpArcGISAuthenticationChallengeHandler(oAuthUserSignInViewModel)
-
-        // check if the portal can be loaded
-        lifecycleScope.launch {
-            portal.load().onSuccess {
-                MaterialAlertDialogBuilder(this@MainActivity).setMessage(
-                    "Portal succeeded to load, portal user: ${portal.user?.username}"
-                ).show()
-                // authentication complete, display PortalItem
-                mapView.apply {
-                    visibility = View.VISIBLE
-                    map = ArcGISMap(PortalItem(portal.url))
-                }
-            }.onFailure { throwable ->
-                // authentication failed, display error message
-                activityMainBinding.authFailedMessage.apply {
-                    visibility = View.VISIBLE
-                    text = String.format("Portal failed to load, ${throwable.message}")
-                }
+        setContent {
+            SampleAppTheme {
+                AuthenticateWithOAuthApp()
             }
         }
     }
 
-    /**
-     * Sets up the [ArcGISAuthenticationChallengeHandler] to create an
-     * [OAuthUserCredential] by launching a browser page to perform a OAuth user
-     * login prompt using [oAuthUserSignInViewModel]
-     */
-    private fun setUpArcGISAuthenticationChallengeHandler(oAuthUserSignInViewModel: OAuthUserSignInViewModel) {
-        ArcGISEnvironment.authenticationManager.arcGISAuthenticationChallengeHandler =
-            ArcGISAuthenticationChallengeHandler { challenge ->
-                if (oAuthConfiguration.canBeUsedForUrl(challenge.requestUrl)) {
-                    val oAuthUserCredential =
-                        OAuthUserCredential.create(oAuthConfiguration) { oAuthUserSignIn ->
-                            oAuthUserSignInViewModel.promptForOAuthUserSignIn(oAuthUserSignIn)
-                        }.getOrThrow()
+    @Composable
+    private fun AuthenticateWithOAuthApp() {
 
-                    ArcGISAuthenticationChallengeResponse.ContinueWithCredential(oAuthUserCredential)
-                } else {
-                    ArcGISAuthenticationChallengeResponse.ContinueAndFailWithError(
-                        UnsupportedOperationException()
-                    )
-                }
-            }
+        // create a ViewModel to handle interactions
+        val mapViewModel: MapViewModel = viewModel()
+
+        Surface(
+            color = MaterialTheme.colorScheme.background
+        ) {
+            MainScreen(
+                sampleName = getString(R.string.app_name)
+            )
+            // Displays appropriate Authentication UI when an authentication challenge is issued.
+            // Because the authenticatorState has an oAuthUserConfiguration set, authentication
+            // challenges will happen via OAuth.
```

Comment:
I suggest to add here some explanation about where in your app `DialogAuthenticator` should be called, something like:
```suggestion
            // challenges will happen via OAuth.
            // Call the DialogAuthenticator composable function at the top level of your view hierarchy, for example
            // at the same level as MainScreen(). This ensures that authentication handling is set up before any
            // components of the ArcGIS Maps SDK that may require authentication are used.
```